vogue/server/utils/utils.py

```diff
@@ -644,47 +644,58 @@ def microsalt_get_qc_time(adapter, year: int, metric_path: str) -> dict:
             'year': {
                 '$eq': int(year)
             },
-            metric: {'$not' :{'$type': 2}}
+            metric: {'$not': {'$type': 2}}
         }
     }, {
         '$group': {
             '_id': {
                 'month': '$month'
             },
             metric: {
-                '$push': '$' + metric
+                '$push': f"${metric}"
+            },
+            'samples': {
+                '$push': {
+                    'insert_size': '$insert_size',
+                    'id': '$_id'
+                }
             }
         }
     }]
 
     aggregate_result = adapter.bioinfo_samples_aggregate(pipe)
     intermediate = {
-        result['_id']['month']: result[metric]
+        result['_id']['month']: {metric: result[metric], 'samples': result['samples']}
         for result in aggregate_result
     }
     box_plots = []
     labels = []
     means = []
+    outliers = []
     for m in MONTHS:
-        values = intermediate.get(m[0], [])
-        if values:
-            minimum = round(min(values), 2)
-            low = round(np.percentile(values, 25), 2)
-            med = round(np.median(values), 2)
-            high = round(np.percentile(values, 75), 2)
-            maximum = round(max(values), 2)
+        data = intermediate.get(m[0], [])
+        if data:
+            values = data[metric]
+            samples = data['samples']
+            q1 = round(np.percentile(values, 25), 2)
+            q2 = round(np.median(values), 2)
+            q3 = round(np.percentile(values, 75), 2)
+            iqr = q3 - q1
+            maximum = q3 + iqr * 0.5
+            minimum = q1 - iqr * 0.5
+            outliers.extend([{'x': m[0] - 1, 'y': s[metric], 'name': s['id']} for s in samples if
+                             minimum > s[metric] or s[metric] > maximum])
```

Comment:
Suggestion for a bit more descriptive variable names and compact quantile calculation. 50th percentile is median:

```suggestion
            q_25, q_50, q_75 = round(np.percentile(values, [25, 50, 75]), 2)
            iqr = q_75 - q_25
            maximum = q_75 + iqr * 0.5
            minimum = q_25 - iqr * 0.5
            outliers.extend([{'x': m[0] - 1, 'y': s[metric], 'name': s['id']} for s in samples if
                             minimum > s[metric] or s[metric] > maximum])
```

Reply:
It doesnt matter to me relly, but I think its rather standard to call the quartiles Q1, Q2, Q3..

Reply:
Of course! Just a suggestion to harmonize.